### ISSUE TITLE
ImportError: IfcOpenShell not built for python_distribution | Conda, Docker

### ISSUE THREAD

**Issue by eriadam** [IfcOpenShell/IfcOpenShell#629]
I am trying to use the python wrapper in a Docker container installed from Conda. Regardless of the python version used the error message is the same:` "ImportError: IfcOpenShell not built for 'linux/64bit/python...'"`

<details>
<summary>Here is the full output of the packages installed and the stack trace.</summary>
<p>

```
Sending build context to Docker daemon  1.199GB
Step 1/3 : FROM continuumio/miniconda3
 ---> 6b5cf97566c3
Step 2/3 : RUN conda install   -c conda-forge   -c dlr-sc   -c pythonocc   -c oce   -c ifcopenshell   pythonocc-core   ifcopenshell
 ---> Running in cf886eda02bf
Collecting package metadata: ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /opt/conda

  added / updated specs:
    - ifcopenshell
    - pythonocc-core


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    asn1crypto-0.24.0          |        py36_1003         154 KB  conda-forge
    bzip2-1.0.6                |    h14c3975_1002         415 KB  conda-forge
    ca-certificates-2019.6.16  |       hecc5488_0         145 KB  conda-forge
    certifi-2019.3.9           |           py36_0         149 KB  conda-forge
    cffi-1.12.3                |   py36h8022711_0         218 KB  conda-forge
    chardet-3.0.4              |        py36_1003         190 KB  conda-forge
    conda-4.6.14               |           py36_0         2.1 MB  conda-forge
    cryptography-2.7           |   py36h72c5cf5_0         607 KB  conda-forge
    dbus-1.13.2                |       h714fa37_1         554 KB
    expat-2.2.5                |    hf484d3e_1002         185 KB  conda-forge
    fontconfig-2.13.1          |    he4413a7_1000         327 KB  conda-forge
    freeimageplus-3.18.0       |       hf484d3e_2         5.4 MB  dlr-sc
    freetype-2.10.0            |       he983fc9_0         885 KB  conda-forge
    gettext-0.19.8.1           |    hc5be6a0_1002         3.6 MB  conda-forge
    gl2ps-1.3.8                |       h14c3975_4         214 KB  dlr-sc
    glib-2.56.2                |    had28632_1001         4.7 MB  conda-forge
    gst-plugins-base-1.14.0    |       hbbd80ab_1         6.3 MB
    gstreamer-1.14.0           |       hb453b48_1         3.8 MB
    icu-58.2                   |    hf484d3e_1000        22.6 MB  conda-forge
    idna-2.8                   |        py36_1000         132 KB  conda-forge
    ifcopenshell-0.6.0a0       |           py36_0         3.5 MB  ifcopenshell
    jpeg-9c                    |    h14c3975_1001         251 KB  conda-forge
    libboost-1.67.0            |       h46d08c1_4        20.9 MB
    libiconv-1.15              |    h516909a_1005         2.0 MB  conda-forge
    libpng-1.6.37              |       hed695b0_0         343 KB  conda-forge
    libuuid-2.32.1             |    h14c3975_1000          26 KB  conda-forge
    libxcb-1.13                |    h14c3975_1002         396 KB  conda-forge
    libxml2-2.9.9              |       h13577e0_0         2.0 MB  conda-forge
    oce-0.17.3                 |                1        28.6 MB  oce
    openssl-1.1.1b             |       h14c3975_1         4.0 MB  conda-forge
    pcre-8.43                  |       he6710b0_0         260 KB
    pip-19.1.1                 |           py36_0         1.8 MB  conda-forge
    pthread-stubs-0.4          |    h14c3975_1001           5 KB  conda-forge
    pycosat-0.6.3              |py36h14c3975_1001         104 KB  conda-forge
    pycparser-2.19             |           py36_1         173 KB  conda-forge
    pyopenssl-19.0.0           |           py36_0          81 KB  conda-forge
    pyqt-5.6.0                 |py36h13b7fb3_1008         5.4 MB  conda-forge
    pysocks-1.7.0              |           py36_0          26 KB  conda-forge
    python-3.6.8               |       h0371630_0        34.4 MB
    pythonocc-core-0.17.3      |           py36_0        43.3 MB  pythonocc
    qt-5.6.3                   |       h8bf5577_3        45.7 MB
    requests-2.22.0            |           py36_0          84 KB  conda-forge
    ruamel_yaml-0.15.71        |py36h14c3975_1000         257 KB  conda-forge
    setuptools-41.0.1          |           py36_0         612 KB  conda-forge
    sip-4.18.1                 |py36hf484d3e_1000         277 KB  conda-forge
    six-1.12.0                 |        py36_1000          22 KB  conda-forge
    tbb-2019.7                 |       hc9558a2_0         1.4 MB  conda-forge
    urllib3-1.24.3             |           py36_0         151 KB  conda-forge
    wheel-0.33.4               |           py36_0          34 KB  conda-forge
    xorg-libxau-1.0.9          |       h14c3975_0          13 KB  conda-forge
    xorg-libxdmcp-1.1.3        |       h516909a_0          18 KB  conda-forge
    ------------------------------------------------------------
                                           Total:       248.7 MB

The following NEW packages will be INSTALLED:

  bzip2              conda-forge/linux-64::bzip2-1.0.6-h14c3975_1002
  dbus               pkgs/main/linux-64::dbus-1.13.2-h714fa37_1
  expat              conda-forge/linux-64::expat-2.2.5-hf484d3e_1002
  fontconfig         conda-forge/linux-64::fontconfig-2.13.1-he4413a7_1000
  freeimageplus      dlr-sc/linux-64::freeimageplus-3.18.0-hf484d3e_2
  freetype           conda-forge/linux-64::freetype-2.10.0-he983fc9_0
  gettext            conda-forge/linux-64::gettext-0.19.8.1-hc5be6a0_1002
  gl2ps              dlr-sc/linux-64::gl2ps-1.3.8-h14c3975_4
  glib               conda-forge/linux-64::glib-2.56.2-had28632_1001
  gst-plugins-base   pkgs/main/linux-64::gst-plugins-base-1.14.0-hbbd80ab_1
  gstreamer          pkgs/main/linux-64::gstreamer-1.14.0-hb453b48_1
  icu                conda-forge/linux-64::icu-58.2-hf484d3e_1000
  ifcopenshell       ifcopenshell/linux-64::ifcopenshell-0.6.0a0-py36_0
  jpeg               conda-forge/linux-64::jpeg-9c-h14c3975_1001
  libboost           pkgs/main/linux-64::libboost-1.67.0-h46d08c1_4
  libiconv           conda-forge/linux-64::libiconv-1.15-h516909a_1005
  libpng             conda-forge/linux-64::libpng-1.6.37-hed695b0_0
  libuuid            conda-forge/linux-64::libuuid-2.32.1-h14c3975_1000
  libxcb             conda-forge/linux-64::libxcb-1.13-h14c3975_1002
  libxml2            conda-forge/linux-64::libxml2-2.9.9-h13577e0_0
  oce                oce/linux-64::oce-0.17.3-1
  pcre               pkgs/main/linux-64::pcre-8.43-he6710b0_0
  pthread-stubs      conda-forge/linux-64::pthread-stubs-0.4-h14c3975_1001
  pyqt               conda-forge/linux-64::pyqt-5.6.0-py36h13b7fb3_1008
  pythonocc-core     pythonocc/linux-64::pythonocc-core-0.17.3-py36_0
  qt                 pkgs/main/linux-64::qt-5.6.3-h8bf5577_3
  sip                conda-forge/linux-64::sip-4.18.1-py36hf484d3e_1000
  tbb                conda-forge/linux-64::tbb-2019.7-hc9558a2_0
  xorg-libxau        conda-forge/linux-64::xorg-libxau-1.0.9-h14c3975_0
  xorg-libxdmcp      conda-forge/linux-64::xorg-libxdmcp-1.1.3-h516909a_0

The following packages will be UPDATED:

  asn1crypto            pkgs/main::asn1crypto-0.24.0-py37_0 --> conda-forge::asn1crypto-0.24.0-py36_1003
  ca-certificates    pkgs/main::ca-certificates-2019.1.23-0 --> conda-forge::ca-certificates-2019.6.16-hecc5488_0
  cffi                pkgs/main::cffi-1.12.2-py37h2e261b9_1 --> conda-forge::cffi-1.12.3-py36h8022711_0
  chardet                   pkgs/main::chardet-3.0.4-py37_1 --> conda-forge::chardet-3.0.4-py36_1003
  cryptography       pkgs/main::cryptography-2.6.1-py37h1b~ --> conda-forge::cryptography-2.7-py36h72c5cf5_0
  idna                           pkgs/main::idna-2.8-py37_0 --> conda-forge::idna-2.8-py36_1000
  pip                          pkgs/main::pip-19.0.3-py37_0 --> conda-forge::pip-19.1.1-py36_0
  pycosat            pkgs/main::pycosat-0.6.3-py37h14c3975~ --> conda-forge::pycosat-0.6.3-py36h14c3975_1001
  pycparser                pkgs/main::pycparser-2.19-py37_0 --> conda-forge::pycparser-2.19-py36_1
  pysocks                   pkgs/main::pysocks-1.6.8-py37_0 --> conda-forge::pysocks-1.7.0-py36_0
  requests                pkgs/main::requests-2.21.0-py37_0 --> conda-forge::requests-2.22.0-py36_0
  ruamel_yaml        pkgs/main::ruamel_yaml-0.15.46-py37h1~ --> conda-forge::ruamel_yaml-0.15.71-py36h14c3975_1000
  setuptools            pkgs/main::setuptools-41.0.0-py37_0 --> conda-forge::setuptools-41.0.1-py36_0
  six                          pkgs/main::six-1.12.0-py37_0 --> conda-forge::six-1.12.0-py36_1000
  urllib3                  pkgs/main::urllib3-1.24.1-py37_0 --> conda-forge::urllib3-1.24.3-py36_0
  wheel                      pkgs/main::wheel-0.33.1-py37_0 --> conda-forge::wheel-0.33.4-py36_0

The following packages will be SUPERSEDED by a higher-priority channel:

  certifi                pkgs/main::certifi-2019.3.9-py37_0 --> conda-forge::certifi-2019.3.9-py36_0
  conda                      pkgs/main::conda-4.6.14-py37_0 --> conda-forge::conda-4.6.14-py36_0
  openssl              pkgs/main::openssl-1.1.1b-h7b6447c_1 --> conda-forge::openssl-1.1.1b-h14c3975_1
  pyopenssl              pkgs/main::pyopenssl-19.0.0-py37_0 --> conda-forge::pyopenssl-19.0.0-py36_0

The following packages will be DOWNGRADED:

  python                                   3.7.3-h0371630_0 --> 3.6.8-h0371630_0


Proceed ([y]/n)?

Downloading and Extracting Packages
libxcb-1.13          | 396 KB    | ########## | 100%
gettext-0.19.8.1     | 3.6 MB    | ########## | 100%
pycosat-0.6.3        | 104 KB    | ########## | 100%
libiconv-1.15        | 2.0 MB    | ########## | 100%
pycparser-2.19       | 173 KB    | ########## | 100%
wheel-0.33.4         | 34 KB     | ########## | 100%
python-3.6.8         | 34.4 MB   | ########## | 100%
pthread-stubs-0.4    | 5 KB      | ########## | 100%
jpeg-9c              | 251 KB    | ########## | 100%
ca-certificates-2019 | 145 KB    | ########## | 100%
pyopenssl-19.0.0     | 81 KB     | ########## | 100%
dbus-1.13.2          | 554 KB    | ########## | 100%
gst-plugins-base-1.1 | 6.3 MB    | ########## | 100%
ifcopenshell-0.6.0a0 | 3.5 MB    | ########## | 100%
pysocks-1.7.0        | 26 KB     | ########## | 100%
asn1crypto-0.24.0    | 154 KB    | ########## | 100%
idna-2.8             | 132 KB    | ########## | 100%
ruamel_yaml-0.15.71  | 257 KB    | ########## | 100%
pyqt-5.6.0           | 5.4 MB    | ########## | 100%
gl2ps-1.3.8          | 214 KB    | ########## | 100%
oce-0.17.3           | 28.6 MB   | ########## | 100%
freeimageplus-3.18.0 | 5.4 MB    | ########## | 100%
gstreamer-1.14.0     | 3.8 MB    | ########## | 100%
openssl-1.1.1b       | 4.0 MB    | ########## | 100%
setuptools-41.0.1    | 612 KB    | ########## | 100%
bzip2-1.0.6          | 415 KB    | ########## | 100%
tbb-2019.7           | 1.4 MB    | ########## | 100%
xorg-libxdmcp-1.1.3  | 18 KB     | ########## | 100%
six-1.12.0           | 22 KB     | ########## | 100%
freetype-2.10.0      | 885 KB    | ########## | 100%
certifi-2019.3.9     | 149 KB    | ########## | 100%
sip-4.18.1           | 277 KB    | ########## | 100%
conda-4.6.14         | 2.1 MB    | ########## | 100%
libpng-1.6.37        | 343 KB    | ########## | 100%
cryptography-2.7     | 607 KB    | ########## | 100%
icu-58.2             | 22.6 MB   | ########## | 100%
expat-2.2.5          | 185 KB    | ########## | 100%
glib-2.56.2          | 4.7 MB    | ########## | 100%
pythonocc-core-0.17. | 43.3 MB   | ########## | 100%
libxml2-2.9.9        | 2.0 MB    | ########## | 100%
cffi-1.12.3          | 218 KB    | ########## | 100%
libboost-1.67.0      | 20.9 MB   | ########## | 100%
pip-19.1.1           | 1.8 MB    | ########## | 100%
qt-5.6.3             | 45.7 MB   | ########## | 100%
fontconfig-2.13.1    | 327 KB    | ########## | 100%
libuuid-2.32.1       | 26 KB     | ########## | 100%
xorg-libxau-1.0.9    | 13 KB     | ########## | 100%
requests-2.22.0      | 84 KB     | ########## | 100%
pcre-8.43            | 260 KB    | ########## | 100%
chardet-3.0.4        | 190 KB    | ########## | 100%
urllib3-1.24.3       | 151 KB    | ########## | 100%
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Removing intermediate container cf886eda02bf
 ---> 7efa8afeef15
Step 3/3 : COPY . /app
 ---> d75529714ea1
Successfully built d75529714ea1

(base) root@9f42a5e963fe:/app# python main.py
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/ifcopenshell/ifcopenshell_wrapper.py", line 14, in swig_import_helper
    return importlib.import_module(mname)
  File "/opt/conda/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 658, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 571, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 922, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: libboost_system.so.1.65.1: cannot open shared object file: No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/ifcopenshell/__init__.py", line 47, in <module>
    from . import ifcopenshell_wrapper
  File "/opt/conda/lib/python3.6/site-packages/ifcopenshell/ifcopenshell_wrapper.py", line 17, in <module>
    _ifcopenshell_wrapper = swig_import_helper()
  File "/opt/conda/lib/python3.6/site-packages/ifcopenshell/ifcopenshell_wrapper.py", line 16, in swig_import_helper
    return importlib.import_module('_ifcopenshell_wrapper')
  File "/opt/conda/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ModuleNotFoundError: No module named '_ifcopenshell_wrapper'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "main.py", line 1, in <module>
    from IFC.Model import Model
  File "/app/IFC/Model.py", line 1, in <module>
    import ifcopenshell
  File "/opt/conda/lib/python3.6/site-packages/ifcopenshell/__init__.py", line 55, in <module>
    raise ImportError("IfcOpenShell not built for '%s'" % python_distribution)
ImportError: IfcOpenShell not built for 'linux/64bit/python3.6'
```

</p>
</details>

**Comment by aothms** [comment]
Thanks for reporting, apparently there is an issue with the recipe.

> The following packages will be downloaded:
> ...
> libboost-**1.67.0**            |       h46d08c1_4        20.9 MB

> ImportError: libboost_system.so.**1.65**.1: cannot open shared object file: No such file or directory

I am not sure how easy it is to fix this. ifcopenshell is going to be on conda-forge https://github.com/conda-forge/staged-recipes/pull/8277 so it's not worth it to spend a lot of effort fixing this.

I'd say your options are:
* Just pick an appropriate statically linked module and unzip into the site-packages E.g. https://github.com/IfcOpenBot/IfcOpenShell/commit/12e9f7b941e0fca8d4f5ee97e5204c2c5c9eb30c#commitcomment-33953852
* Wait for the conda-forge inclusion
* Somehow try to fix the boost version incompatibility (perhaps 1.65.1 can be installed explicitly before installing ifcopenshell).

**Comment by eriadam** [comment]
Thank you @aothms, forcing the specific version of boost did solve the problem.

```
RUN conda install \
  -c conda-forge \
  -c dlr-sc \
  -c pythonocc \
  -c oce \
  -c ifcopenshell \
  libboost=1.65.1 \
  pythonocc-core \
  ifcopenshell
```

**Comment by cassie100** [comment]
i have no idea what your talkin bout 

    On Wednesday, June 19, 2019, 6:34:55 AM EDT, Adam Eri <notifications@github.com> wrote:  
 
 
Closed #629.
